### PR TITLE
[WIP] stdio option handling

### LINKED ISSUE #24: Better `stdio` option.
I think the `stdio` array is a pretty poor UI, you have to remember the order of things. I'm constantly pulling up that piece of documentation. Instead, of the three member array, let's just give each one it's own named property:

``` js
execa('./cli.js', [], {
  stdin: someReadableStream,
  stdout: 'pipe',
  stderr: 'inherit'
});
```

### PR DESCRIPTION
**Pull request by jamestalmage**
Fixes #24.

I've only written an options helper method. Not integrated into index.js yet. Started this before #27, and it's backburnered until we get some of the remaining issues from that PR resolved. I'll pick this back up once we've made progress on those.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -34,12 +34,12 @@ test('stdout/stderr available on errors', async t => {
...
 
 test('include stdout in errors for improved debugging', async t => {
-	const err = await t.throws(m('fixtures/error-message.js'));
+	const err = await t.throws(m('../fixtures/error-message.js'));
 	t.regex(err.message, /stdout/);
 });
 
 test('execa.shell()', async t => {
-	const {stdout} = await m.shell('node fixtures/noop foo');
+	const {stdout} = await m.shell('node ../fixtures/noop foo');
```
Comment (jamestalmage) [inline test/index.js]: Required because: https://github.com/sindresorhus/ava/issues/32#issuecomment-214591032

File: test/stdio.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+t({stdio: 'inherit'}, 'inherit');
+t({stdio: 'pipe'}, 'pipe');
+t({stdio: 'ignore'}, 'ignore');
+
+t({}, [null, null, null]);
+t({stdio: []}, [null, null, null]);
+t({stdin: 'pipe'}, ['pipe', null, null]);
+t({stdout: 'ignore'}, [null, 'ignore', null]);
+t({stderr: 'inherit'}, [null, null, 'inherit']);
```
Comment (sindresorhus) [inline test/stdio.js]: Can you also do one with `stdout` and `stderr` at the same time?

File: test/stdio.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	test(util.inspect(input, {colors: true}), t => {
+		const result = stdio(input);
+
+		if (typeof expected === 'object' && expected !== null) {
+			t.deepEqual(result, expected);
+		} else {
+			t.is(result, expected);
+		}
+	});
+}
```
Comment (sindresorhus) [inline test/stdio.js]: We really need test macros.

File: test/stdio.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+t({stdio: 'ignore'}, 'ignore');
+
+t({}, [null, null, null]);
+t({stdio: []}, [null, null, null]);
+t({stdin: 'pipe'}, ['pipe', null, null]);
+t({stdout: 'ignore'}, [null, 'ignore', null]);
+t({stderr: 'inherit'}, [null, null, 'inherit']);
+
+// precedence
+t({stdin: 'inherit', stdio: 'pipe'}, 'pipe');
```
Comment (sindresorhus) [inline test/stdio.js]: Couldn't we just throw if they're used together?

File: lib/stdio.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	}
+
+	if (typeof opts.stdio === 'string') {
+		return opts.stdio;
+	}
+
+	var stdio = opts.stdio || [];
+
+	if (!Array.isArray(stdio)) {
+		throw new TypeError('Incorrect value of stdio option: ' + util.inspect(stdio));
```
Comment (sindresorhus) [inline lib/stdio.js]:
```
`stdio`
```

File: test/stdio.js (hunk 4 above)
Comment (jamestalmage) [inline test/stdio.js]: Yeah, that's probably the way to go.

**Comment by SamVerschueren** [comment]
Is there any way I could take over this PR? Or should @jamestalmage give me write access to his repository?

**Comment by sindresorhus** [comment]
@SamVerschueren Just create a new branch in this repo, cherrypick his commit, do your changes, and open a new PR.

**Comment by jamestalmage** [comment]
Closing in favor of #80

**Pull request closed without merging**